src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= this.table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = calculateIndex(key, table.length);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if ((key == null && current.key == null)
+                        || key != null && key.equals(current.key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = new Node<>(key, value);
+                    size++;
+                    return;
+                }
+                current = current.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> current = table[calculateIndex(key, table.length)];
+        while (current != null) {
+            if ((key == null && current.key == null)
```

Comment:
```suggestion
            if ((key == null && current.key == null)
```
not a mistake but you can remove redundant parentheses if you wish, and condition can be simplified
```suggestion
            if (key == current.key
```